src/main/java/core/basesyntax/service/impl/ReportGenerator.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.RecordDao;
+import core.basesyntax.dao.RecordDaoImpl;
+import core.basesyntax.model.Product;
+import java.util.List;
+
+public class ReportGenerator implements core.basesyntax.service.ReportGenerator {
+    private static final String QUANTITY = "quantity";
+    private static final String COMMA = ",";
+    private final RecordDao recordDao = new RecordDaoImpl();
+
+    @Override
+    public String generateReport() {
+        List<Product> products = recordDao.getAllFromDB();
+        if (products.isEmpty()) {
+            throw new RuntimeException("Can't read products from db");
+        }
+        String header = products.get(0).getClass().getSimpleName().toLowerCase() + COMMA + QUANTITY;
```

Comment:
What if products are of different types, not only Fruits, but Vegetables for example?